File: sortings/Bubblesort/Bubblesort.cpp
#include<iostream>
#include<iostream>
#include<vector>

using namespace std;

vector<int> BubbleSort(vector<int> arr){
    for(int i=0;i<arr.size();i++){
        for(int j=0;j<arr.size()-1;j++){
            if(arr[j+1]<arr[j]){
                int temp = arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
            }
        }
    }
    return arr;
}

int main(){
    vector<int> nums = {995,123,520,420,620,450};
    vector<int> res = BubbleSort(nums);
    for(int i:res) cout << i << " ";
    return 0;
}

File: sortings/Bubblesort/JS/Bubblesort.js
//Array to sort in incremental order
const arr = [95, 23, 45, 61, 10];

function Bubblesort(arr) {
  for (let i = 0; i < arr.length; i++) {
    // First we start to iterate for lenght times
    for (let j = 0; j < arr.length - i - 1; j++) {
      // we start another loop which value is less than the arr.lenght-i-1 because the bigger value already moved to the last of the array
      if (arr[j] > arr[j + 1]) {
        // check if the adjacent values
        temp = arr[j]; // if before value is higher then swap
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }

  return arr;
}

console.log(Bubblesort(arr));
